libs/server/Resp/RespCommandInfoParser.cs

```diff
@@ -0,0 +1,439 @@
+                                    out var flag)) return false;
+                            flags |= flag;
+                        }
+
+                        break;
+                    case "begin_search":
+                        if (!RespKeySpecificationTypesParser.TryReadFromResp(ksKey, ref ptr, end, out beginSearch)) return false;
+                        break;
+                    case "find_keys":
+                        if (!RespKeySpecificationTypesParser.TryReadFromResp(ksKey, ref ptr, end, out findKeys)) return false;
+                        break;
+                }
+            }
+
+            keySpec = new RespCommandKeySpecification()
+            {
+                Notes = notes,
+                Flags = flags,
+                BeginSearch = beginSearch,
+                FindKeys = findKeys
+            };
+
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Logic for parsing BeginSearch / FindKeys key specification from RESP format
+    /// </summary>
+    internal class RespKeySpecificationTypesParser
+    {
+        /// <summary>
+        /// Tries to parse KeySpecBase from RESP format
+        /// </summary>
+        /// <param name="keySpecKey">Type of key specification ("begin_search" / "find_keys")</param>
+        /// <param name="ptr">Pointer to current RESP chunk to read</param>
+        /// <param name="end">Pointer to end of RESP chunk to read</param>
+        /// <param name="keySpec">Parsed KeySpecBase object</param>
+        /// <returns>True if parsing successful</returns>
+        public static unsafe bool TryReadFromResp(string keySpecKey, ref byte* ptr, byte* end, out KeySpecBase keySpec)
+        {
+            keySpec = default;
+
+            if (!TryReadKeySpecHeader(ref ptr, end, out var keySpecType)) return false;
+
+            IKeySpecParser parser = (keySpecKey, keySpecType) switch
+            {
+                ("begin_search", "index") => BeginSearchIndexParser.Instance,
+                ("begin_search", "keyword") => BeginSearchKeywordParser.Instance,
+                ("begin_search", "unknown") => BeginSearchUnknownParser.Instance,
+                ("find_keys", "range") => FindKeysRangeParser.Instance,
+                ("find_keys", "keynum") => FindKeysKeyNumParser.Instance,
+                ("find_keys", "unknown") => FindKeysUnknownParser.Instance,
+                _ => null
+            };
+
+            if (parser == null || !parser.TryReadFromResp(ref ptr, end, out keySpec)) return false;
+
+            return true;
+        }
+
+        private static unsafe bool TryReadKeySpecHeader(ref byte* ptr, byte* end, out string keySpecType)
+        {
+            keySpecType = default;
+
+            RespReadUtils.ReadArrayLength(out var ksTypeElemCount, ref ptr, end);
+            if (ksTypeElemCount != 4) return false;
+            RespReadUtils.ReadStringWithLengthHeader(out var ksTypeStr, ref ptr, end);
+            if (ksTypeStr != "type") return false;
```

Comment:
nit: A bit faster & "precise" comparisons with `StringComparison.Ordinal` which wont allow any funny business by culture normalizations etc.
```suggestion
            if (!ksTypeStr.Equals("type", StringComparison.Ordinal)) return false;
```

ditto for all string comparisongs in RESP parsing :+1:

Reply:
Looking back at this, I was confused.. The `string.Equals` and the equality operators already do Ordinal comparison. But it's still [good practice](https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings) to explicitly specify the `CultureInfo` as this is very inconsistent across all the methods..

![image](https://github.com/microsoft/garnet/assets/26825431/ea854223-c4bf-402b-9b33-465412ed0c5c)